query/src/sql/planner/binder/scalar.rs

```diff
@@ -48,6 +52,74 @@ impl ScalarBinder {
             Expr::BinaryOp { op, left, right } => {
                 self.bind_binary_op(op, left.as_ref(), right.as_ref(), bind_context)
             }
+            Expr::FunctionCall {
+                distinct,
+                name,
+                args,
+                params,
+            } => {
+                match AggregateFunctionFactory::instance().check(name) {
+                    true => {
+                        // Function is aggregate function
+                        let mut data_values = Vec::with_capacity(params.len());
+                        for param in params.iter() {
+                            data_values.push(match param {
+                                Literal::Number(val) => DataValue::try_from_literal(val, None)?,
+                                Literal::String(val) => DataValue::String(val.clone().into_bytes()),
+                                Literal::Boolean(val) => DataValue::Boolean(*val),
+                                Literal::Null => DataValue::Null,
+                            })
+                        }
+
+                        let scalar_binder = ScalarBinder::new();
+                        let mut scalar_exprs = Vec::with_capacity(args.len());
+                        for arg in args.iter() {
+                            scalar_exprs.push(
+                                scalar_binder
+                                    .bind_expr(arg, bind_context)?
+                                    .as_any()
```

Comment:
maybe move `.as_any().downcast_ref.ok_or_else` into `ScalarExprRef` is better?

```
impl ScalarExpr {
    pub fn safe_cast<To>(&self) -> Result<To> {
        self.as_any().downcast_ref::<To>().ok_or_else(|| {...})
    }
}
```

Reply:
Nice idea!